File: 백준/Silver/14501. 퇴사/퇴사.py
import sys
input = sys.stdin.readline

n = int(input())
counseling = []
for i in range(n):
    counseling.append(tuple(map(int, input().split())))

dp = [0] * (n+1)

for i in range(n):
    for j in range(i + counseling[i][0], n+1):
        if dp[j] < dp[i] + counseling[i][1]:
            dp[j] = dp[i] + counseling[i][1]

print(dp[-1])

File: 백준/Silver/14501. 퇴사/퇴사.js
const fs = require('fs')

let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')
const n = parseInt(input[0].trim());
const counseling = [];

let i = 0;
let j = 0;

for (i=1; i<input.length; i++) {
    let tmp = input[i].split(' ');
    for (j=0; j<2; j++){
        tmp[j] = parseInt(tmp[j].trim());
    }
    counseling.push(tmp);
}

let dp = [];
for (i=0; i<n+1; i++){
    dp.push(0);
}

for (i=0; i<n; i++){
    for (j=i+counseling[i][0]; j<n+1; j++){
        if (dp[j] < dp[i] + counseling[i][1]){
            dp[j] = dp[i] + counseling[i][1]
        }
    }
}

console.log(dp[n]);